### PR TITLE
Make bootstrap json test case more robust

### PR DESCRIPTION
**Pull request by ixdy**
Small fixes after #2119.

### CODE REVIEW

File: jenkins/bootstrap_test.py [hunk 1]
```diff
@@ -1499,10 +1500,9 @@ def Check(job, name):
             self.assertIn('frequency', job)
             self.assertIn('repo-name', job)
             self.assertIn('timeout', job)
-            modern = is_modern(name)  # TODO(fejta): all jobs
-            self.assertEquals(modern, job.get('json', 1), name)
-            if is_modern(name):  # TODO(fejta): do this for all jobs
-                self.assertGreater(job['timeout'], 0, name)
+            self.assertNotIn('json', job)
+            self.assertTrue(is_modern(name))
```
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: I think you can get rid of `is_modern` here now

File: jenkins/bootstrap_test.py [hunk 2]
```diff
@@ -1499,10 +1500,9 @@ def Check(job, name):
             self.assertIn('frequency', job)
             self.assertIn('repo-name', job)
             self.assertIn('timeout', job)
-            modern = is_modern(name)  # TODO(fejta): all jobs
-            self.assertEquals(modern, job.get('json', 1), name)
-            if is_modern(name):  # TODO(fejta): do this for all jobs
-                self.assertGreater(job['timeout'], 0, name)
+            self.assertNotIn('json', job)
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Nice

File: jenkins/bootstrap_test.py (hunk 1 above)
Comment (fejta) [inline jenkins/bootstrap_test.py]: Yeah modern == --json so hopefully everything is modern now :)

File: jenkins/bootstrap_test.py [hunk 3]
```diff
@@ -1462,7 +1462,8 @@ def Check(job, name):
             self.assertIn('trigger-job', job)
             self.assertNotIn('branch', job)
             modern = is_modern(name)
-            self.assertEquals(modern, job.get('json', 1), name)
+            # Default value for jobs is now 1
+            self.assertEqual(modern, job.get('json', 1), name)
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: I like the not in pattern better.

File: jenkins/bootstrap_test.py (hunk 1 above)
Comment (ixdy) [inline jenkins/bootstrap_test.py]: Removed.

File: jenkins/bootstrap_test.py (hunk 3 above)
Comment (ixdy) [inline jenkins/bootstrap_test.py]: test case is now even better - `json: 1` is disallowed.

**Comment by krzyzacy** [comment]
Now you can remove `is_modern = lambda n: '-e2e-' not in n`, and pass `use_json=True` in `CheckBootstrapYaml`

**Comment by ixdy** [comment]
done

**Pull request merged**